### ISSUE TITLE
[BUG] Numeric Indexes incorrectly compare as equal to numpy dtype objects 

### ISSUE THREAD

**Issue by brandon-b-miller** [rapidsai/cudf#8895]
**Describe the bug**
Numeric indicies compare as true using ` == ` against numpy dtypes. 

**Steps/Code to reproduce bug**

```python
cudf.Int8Index([]) == np.dtype('int8')
# True
cudf.Int8Index([]) == np.dtype('int16')
# False
```

**Expected behavior**
Either this should return False or it should return `[False] * len(the_index)` (pandas behavior)

**Environment overview (please complete the following information)**
 - Environment location: Bare-Metal
 - Method of cuDF install: Source

**Environment details**

**Additional context**

**Comment by vyasr** [comment]
@brandon-b-miller was this blocking something in Dask-SQL? I see that it's marked for that.

**Comment by brandon-b-miller** [comment]
No - dask-sql is not blocked by this. cc @randerzander just to make sure.

**Comment by vyasr** [comment]
Assuming it's not a blocker then could you move it to 22.02 once @randerzander confirms?

**Comment by vyasr** [comment]
This issue is no longer relevant because all of the integral index types have been removed as part of the pandas 2.0 migration 24.04.
